src/uct/ib/base/ib_md.h

```diff
@@ -615,7 +610,24 @@ uct_ib_md_handle_mr_list_multithreaded(uct_ib_md_t *md, void *address,
 ucs_status_t uct_ib_reg_key_impl(uct_ib_md_t *md, void *address, size_t length,
                                  uint64_t access_flags, int dmabuf_fd,
                                  size_t dmabuf_offset, uct_ib_mem_t *memh,
-                                 uct_ib_mr_t *mr, uct_ib_mr_type_t mr_type,
+                                 struct ibv_mr **mr_p, uct_ib_mr_type_t mr_type,
                                  int silent);
 
+uint64_t uct_ib_memh_access_flags(uct_ib_md_t *md, uct_ib_mem_t *memh);
```

Comment:
uct_ib_memh_access_flags and uct_ib_memh_new are supposed to be used outside?

Reply:
yes, they will be used from ib_mlx5dv_md.c